src/main/java/teammates/ui/webapi/action/GetFeedbackSessionsAction.java

```diff
@@ -20,12 +23,27 @@ protected AuthType getMinAuthLevel() {
 
     @Override
     public void checkSpecificAccessControl() {
+        if (!(userInfo.isStudent || userInfo.isInstructor)) {
+            throw new UnauthorizedAccessException("Student or instructor privilege is required to access this resource.");
+        }
+
         String courseId = getRequestParamValue(Const.ParamsNames.COURSE_ID);
-        if (courseId == null) {
-            gateKeeper.verifyInstructorPrivileges(logic.getAccount(userInfo.getId()));
+        boolean isRequestAsStudent = getRequestParamValue(Const.ParamsNames.IS_IN_RECYCLE_BIN) == null;
+
+        if (isRequestAsStudent) {
+            if (courseId == null) {
+                gateKeeper.verifyStudentPrivileges(logic.getAccount(userInfo.getId()));
+            } else {
+                CourseAttributes courseAttributes = logic.getCourse(courseId);
+                gateKeeper.verifyAccessible(logic.getStudentForGoogleId(courseId, userInfo.getId()), courseAttributes);
+            }
         } else {
-            CourseAttributes courseAttributes = logic.getCourse(courseId);
-            gateKeeper.verifyAccessible(logic.getInstructorForGoogleId(courseId, userInfo.getId()), courseAttributes);
+            if (courseId == null) {
+                gateKeeper.verifyInstructorPrivileges(logic.getAccount(userInfo.getId()));
```

Comment:
Can just assert isInstructor is true? (and save one DB read `logic.getAccount`)?